UCLIBC_EXTRA_CFLAGS is currently added before the OPTIMIZATION flags
and OPTIMIZATION is chosen to be Os by default. But in OE we pass the optimisation
flags through UCLIBC_EXTRA_CFLAGS but they are not effective since -Os is
specified at last. So we need to change the order of these option flags

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending

diff --git a/Rules.mak b/Rules.mak
index 65fe47c..eca7f40 100644
--- a/Rules.mak
+++ b/Rules.mak
@@ -585,9 +574,6 @@ CFLAGS := -include $(top_srcdir)include/libc-symbols.h \
 	-nostdinc -I$(top_builddir)include -I$(top_srcdir)include -I. \
 	-I$(top_srcdir)libc/sysdeps/linux \
 	-I$(top_srcdir)libc/sysdeps/linux/$(TARGET_ARCH)
-ifneq ($(strip $(UCLIBC_EXTRA_CFLAGS)),"")
-CFLAGS += $(call qstrip,$(UCLIBC_EXTRA_CFLAGS))
-endif
 
 # We need this to be checked within libc-symbols.h
 ifneq ($(HAVE_SHARED),y)
@@ -633,6 +619,9 @@ LDFLAGS += -Wl,-s
 else
 STRIPTOOL := true -Stripping_disabled
 endif
+ifneq ($(strip $(UCLIBC_EXTRA_CFLAGS)),"")
+CFLAGS += $(call qstrip,$(UCLIBC_EXTRA_CFLAGS))
+endif
 
 ifeq ($(DOMULTI),y)
 # we try to compile all sources at once into an object (IMA), but
